server/models/video/video-channel-sync.ts

```diff
@@ -0,0 +1,163 @@
+import { Op } from "sequelize"
+import { UserModel } from "../user/user"
+import { isVideoImportTargetUrlValid } from "@server/helpers/custom-validators/video-imports"
+
+type AvailableForListOptions = {
+  accountId: number
+  search?: string
+  host?: string
+  handles?: string[]
+  forCount?: boolean
+}
+
+@Table({
+  tableName: 'videoChannelSync'
+})
+export class VideoChannelSyncModel extends Model<Partial<AttributesOnly<VideoChannelSyncModel>>> {
+
+  @AllowNull(false)
+  @Default(null)
+  @Is('VideoChannelExternalChannelUrl', value => throwIfNotValid(value, isVideoImportTargetUrlValid, 'externalChannelUrl', true))
+  @Column(DataType.STRING(CONSTRAINTS_FIELDS.VIDEO_CHANNEL_SYNCS.EXTERNAL_CHANNEL_URL.max))
+  externalChannelUrl: string
+
+  @CreatedAt
+  createdAt: Date
+
+  // We have no use of this field
+  @UpdatedAt
+  updatedAt: Date
+
+  @ForeignKey(() => VideoChannelModel)
+  @Column
+  videoChannelId: number
+
+  @BelongsTo(() => VideoChannelModel, {
+    foreignKey: {
+      allowNull: false
+    },
+    onDelete: 'cascade'
+  })
+  VideoChannel: VideoChannelModel
+
+  @AllowNull(false)
+  @Default(VideoChannelSyncState.WAITING_FIRST_RUN)
+  @Is('VideoChannelSyncState', value => throwIfNotValid(value, isVideoChannelSyncStateValid, 'state'))
+  @Column
+  state: VideoChannelSyncState
+
+  static listByAccountForAPI (options: Pick<AvailableForListOptions, 'accountId'> & {
+    start: number
+    count: number
+    sort: string
+  }) {
+    const getQuery = (forCount: boolean) => {
+      const accountModel = forCount
+        ? AccountModel.unscoped()
+        : AccountModel
+
+      return {
+        offset: options.start,
+        limit: options.count,
+        order: getSort(options.sort),
+        include: [
+          {
+            model: VideoChannelModel.unscoped(),
+            required: true,
+            include: [
+              {
+                model: accountModel,
+                where: {
+                  id: options.accountId
+                },
+                required: true
+              },
+              {
+                model: ActorModel.unscoped()
+              }
+            ]
+          }
+        ]
+      }
+    }
+
+    return Promise.all([
+      VideoChannelSyncModel.count(getQuery(true)),
+      VideoChannelSyncModel.findAll(getQuery(false))
+    ]).then(([ total, data ]) => ({ total, data }))
+  }
+
+  toFormattedJSON (this: MChannelSyncFormattable): VideoChannelSync {
+    return {
+      id: this.id,
+      state: {
+        id: this.state,
+        label: VIDEO_CHANNEL_SYNC_STATE[this.state]
+      },
+      externalChannelUrl: this.externalChannelUrl,
+      createdAt: this.createdAt.toString(),
+      channel: this.VideoChannel?.toFormattedSummaryJSON(),
+      videoChannelId: this.videoChannelId
+    }
+  }
+
+  static loadById (id: number) {
+    return this.findOne({
+      where: { id }
+    })
+  }
+
+  static async listSyncs (): Promise<VideoChannelSyncModel[]> {
```

Comment:
Please create a dedicated type for the result of this function

For example `VideoChannelSyncChannel` that would be the VideoChannelSync entity with the MVideoChannel association because the caller doesn't seem to need more than the channel name and channel actor id